lib/Jsrt/ChakraCommon.h

```diff
@@ -2063,6 +2072,74 @@ typedef unsigned short uint16_t;
             _Out_ unsigned int *bufferLength);
 
     /// <summary>
+    ///     Creates a Javascript SharedArrayBuffer object with shared content get from JsGetSharedArrayBufferContent.
+    /// </summary>
+    /// <remarks>
+    ///     Requires an active script context.
+    /// </remarks>
+    /// <param name="sharedContents">
+    ///     The storage object of a SharedArrayBuffer which can be shared between multiple thread.
+    /// </param>
+    /// <param name="result">The new SharedArrayBuffer object.</param>
+    /// <returns>
+    ///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+    /// </returns>
+    CHAKRA_API
```

Comment:
>>Just to confirm, one can only create sab from sharedContents returned from a JsGetSharedArrayBufferContent call somewhere, correct?

currently yes, in order to run the test262 tests. we can add API to create SAB if user specify a length and chakra core manage the buffer. if user want to create SAB with arbitrary buffer that will be complicated, the free function to free that buffer will be needed

>>Is the output JsValueRef *result shareable between threads

only JsSharedArrayBufferContent can be shared between threads, and require JsSharedContentAddRef/JsSharedContentRelease call to manage the lifetime. the JsValueRef output of JsCreateSharedArrayBuffer is same as other javascript object which should only be used in current thread

>> A JsCreateExternalSharedArrayBuffer can also be tricky since I don't know if native access to the buffer can break atomics

I don't think that will break atomics as long as the host guarantee that it does not mutate the buffer, this is mostly about memory management, we need a way to notify the host that chakracore has done using the buffer (aka. refcount get to 0). we can discuss about this if you have a use case. for this PR it's not necessary for running test262

I'll change JsCreateSharedArrayBuffer to JsCreateSharedArrayBufferWithSharedContent or you can suggest a better name?